tests/testthat/test-coveralls.R

```diff
@@ -50,105 +50,121 @@ read_file <- function(file) paste(collapse = "\n", readLines(file))
 cov <- package_coverage("TestS4")
 
 test_that("coveralls generates a properly formatted json file", {
-
-  withr::with_envvar(c(ci_vars, "CI_NAME" = "FAKECI"),
-    with_mock(
-      `httr:::RETRY` = function(...) list(...),
-      `httr::content` = identity,
-      `httr::upload_file` = function(file) readChar(file, file.info(file)$size),
-
-      res <- coveralls(coverage = cov),
-      json <- jsonlite::fromJSON(res$body$json_file),
-
-      expect_equal(nrow(json$source_files), 1),
-      expect_equal(json$service_name, "fakeci"),
-      expect_match(json$source_files$name, rex::rex("R", one_of("/", "\\"), "TestS4.R")),
-      expect_equal(json$source_files$source, read_file("TestS4/R/TestS4.R")),
-      expect_equal(json$source_files$source_digest, '1233f2eca5d84704101cb9d9b928f2e9'),
-      expect_equal(json$source_files$coverage[[1]],
-        c(NA, NA, NA, NA, NA, NA, 5, 2, NA, 3, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
-          NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA))
+  withr::with_envvar(
+    c(ci_vars, "CI_NAME" = "FAKECI"),
+    with_mocked_bindings(
```

Comment:
@olivroy if you have a bit more time, I think it would be worth switching these to use to `local_envvar()` and `local_mocked_bindings()` to reduce some of the nesting.